scipy/stats/_multivariate.py

```diff
@@ -437,7 +537,31 @@ def _process_quantiles(self, x, dim):
 
         return x
 
-    def _logpdf(self, x, mean, prec_U, log_det_cov, rank):
+    def _create_cov_info(self, cov, inverse_cov, allow_singular=True):
```

Comment:
Along the lines of https://github.com/scipy/scipy/pull/16002/files#r906859132, I think this logic could go in `_CovInfo`. Similar logic could be added for eigendecomposition or variances and rotation angles.